src/App.tsx

```diff
@@ -127,77 +234,92 @@ export const App: React.FC = () => {
+              <a
+                href="#/"
+                className={`filter__link ${cn({ selected: filter === 'All' })}`}
+                data-cy="FilterLinkAll"
+                onClick={() => setFilter('All')}
+              >
+                All
+              </a>
 
-            <a
-              href="#/completed"
-              className="filter__link"
-              data-cy="FilterLinkCompleted"
-            >
-              Completed
-            </a>
-          </nav>
+              <a
+                href="#/active"
+                className={`filter__link ${cn({ selected: filter === 'Active' })}`}
+                data-cy="FilterLinkActive"
+                onClick={() => setFilter('Active')}
+              >
+                Active
+              </a>
 
-          {/* this button should be disabled if there are no completed todos */}
-          <button
-            type="button"
-            className="todoapp__clear-completed"
-            data-cy="ClearCompletedButton"
-          >
-            Clear completed
-          </button>
-        </footer>
+              <a
+                href="#/completed"
+                className={`filter__link ${cn({ selected: filter === 'Completed' })}`}
+                data-cy="FilterLinkCompleted"
+                onClick={() => setFilter('Completed')}
+              >
+                Completed
+              </a>
+            </nav>
+
+            {/* this button should be disabled if there are no completed todos */}
+            <button
+              type="button"
+              className="todoapp__clear-completed"
+              data-cy="ClearCompletedButton"
+              onClick={() => setTodos(todos.filter(el => !el.completed))}
+            >
+              Clear completed
+            </button>
+          </footer>
+        )}
       </div>
 
       {/* DON'T use conditional rendering to hide the notification */}
       {/* Add the 'hidden' class to hide the message smoothly */}
       <div
         data-cy="ErrorNotification"
-        className="notification is-danger is-light has-text-weight-normal"
+        className={cn({
+          'notification is-danger is-light has-text-weight-normal': errors,
+          hidden: !errors,
+        })}
       >
-        <button data-cy="HideErrorButton" type="button" className="delete" />
-        {/* show only one message at a time */}
-        Unable to load todos
-        <br />
-        Title should not be empty
-        <br />
+        {errors && (
+          <button
+            data-cy="HideErrorButton"
+            type="button"
+            className="delete"
+            onClick={() => setErrors('')}
+          />
+        )}
+        {/* Unable to load todos
+        <br /> */}
+        {errors}
+        {/* <br />
         Unable to add a todo
         <br />
         Unable to delete a todo
         <br />
-        Unable to update a todo
+        Unable to update a todo */}
       </div>
```

Comment:
please split your code on components, because it's very hard to read, even when the program is still quite small.